core/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -327,4 +338,75 @@ private ParsedQuery toQuery(QueryBuilder queryBuilder, CheckedFunction<QueryBuil
     public final Index index() {
         return indexSettings.getIndex();
     }
+
+    /**
+     * Compiles (or retrieves from cache) and executes the provided script
```

Comment:
This doesn't actually execute right? It is the binding that happens immediately (vs lazy below is binding to params later).